torch/csrc/jit/frontend/script_type_parser.cpp

```diff
@@ -271,7 +271,7 @@ std::vector<Argument> ScriptTypeParser::parseArgsFromDecl(
     if (!decl_arg.type().present()) {
       // If this param doesn't have a type, default to "tensor"
       is_inferred_type = true;
```

Comment:
we are already populating `is_inferred_type` here, and passing that information to `Argument`. If we are adding it as metadata to TensorType, then we should remove the old code code that does the same thing.